### PR TITLE
chore(esm): use esm instead of cjs 

### PR DESCRIPTION
**Pull request by AugustinMauroy**

### CODE REVIEW

File: doc/compiler.md [hunk 1]
```diff
@@ -7,25 +7,26 @@ The ProtoDef compiler can convert your protocol JSON into javascript code that c
...
-```javascript
-const ProtoDef = require('protodef').ProtoDef
+```js
+
+import { ProtoDef } from 'protodef'
 
 // Create a ProtoDef instance
 const proto = new ProtoDef()
-proto.addTypes(require('./protocol.json'))
+proto.addTypes(import('./protocol.json'))
```
Comment (IceTank) [inline doc/compiler.md:16]: I think dynamic imports return a promise that wraps the imported value. So this should not work. You would have to refractor the entire file to use async and then use await.

Comment (AugustinMauroy) [reply doc/compiler.md:16]: Yep that an issue

Comment (u9g) [reply doc/compiler.md:16]: this file can be JSON.parse'd from fs.readFileSync

Comment (AugustinMauroy) [reply doc/compiler.md:16]: Also I think FS is the better option

**Pull request closed without merging**